src/adaptations/device-layer/include/Weave/DeviceLayer/PlatformManager.h

```diff
@@ -28,6 +28,19 @@
 
 namespace nl {
 namespace Weave {
+namespace System {
```

Comment:
I believe this be done via appropriate include of the authoritative header file

Reply:
I agree, however, including the authoritative header file (SystemLayer.h) only defines these functions when `WEAVE_SYSTEM_CONFIG_USE_LWIP` is asserted, so when that is not asserted, the build will fail.

Reply:
Upon reflection, I think this is the right location for these forward declarations, since the PlatformManager class chooses to friend these functions regardless of whether they're actually defined/used.

Reply:
Agreed.  I was going to argue that "what if those functions change?  Now we have to change this header as well".  but that's a wrong argument -- we would already have to change this header because the friend declaration repeats the declaration.  With the proposed change, changing the declarations in `SystemLayer.h` would not result in a compilation failure but rather in a link failure subsequently.